### ISSUE TITLE
Models with integrals over an infinite interval have loss = NaN or loss = Inf

### ISSUE THREAD

**Issue by killah-t-cell** [SciML/NeuralPDE.jl#435]
Though we made some fixes in Zygote and Quadrature.jl to add support for integrals with infinite intervals, it seems like NeuralPDE is for some reason still not able to process them. Whenever I create a model with an integral with an infinite interval, the loss function starts at either `NaN` or `Inf` (only in the equations where an integral is present). Of course, some infinite integrals are divergent so Inf can make sense. But `NaN` seems a bit weird.

Take for example the following system. The initial condition are set so that the integral should be approximately = 1 (it describes a distribution function so values don't go past 1 or below 0), but the loss starts at NaN nonetheless.

```julia
@parameters t x v
@variables f(..) E(..) 
Dx = Differential(x)
Dt = Differential(t)
Dv = Differential(v)

v_th = sqrt(2)

# Integrals
Iv = Integral(v in DomainSets.ClosedInterval(-Inf, Inf)) 

eqs = [Dt(f(t,x,v)) ~ - v * Dx(f(t,x,v)) - E(t,x) * Dv(f(t,x,v))
       Dx(E(t,x)) ~ (Iv(f(t,x,v)) - 1)]

bcs = [f(0,x,v) ~ 1/(v_th * sqrt(2π)) * exp(-v^2/(2*v_th^2)),
       E(0,x) ~ (Iv(f(0,x,v)) - 1)]

domains = [t ∈ Interval(0.0, 1.0),
           x ∈ Interval(0.0, 1.0), 
           v ∈ Interval(0.0, 1.0)]

# Neural Network
chain = [FastChain(FastDense(3, 16, Flux.σ), FastDense(16,16,Flux.σ), FastDense(16, 1)),
         FastChain(FastDense(2, 16, Flux.σ), FastDense(16,16,Flux.σ), FastDense(16, 1))]
initθ = map(c -> Float64.(c), DiffEqFlux.initial_params.(chain))

discretization = NeuralPDE.PhysicsInformedNN(chain, QuadratureTraining(), init_params= initθ)
@named pde_system = PDESystem(eqs, bcs, domains, [t,x,v], [f(t,x,v), E(t,x)])
prob = SciMLBase.discretize(pde_system, discretization)

cb = function (p,l)
    println("Current loss is: $l")
    return false
end

# Solve
opt = Optim.BFGS()
res = GalacticOptim.solve(prob, opt, cb = cb, maxiters=5)
```

Some models, however, seem to work. (as you can see here https://github.com/SciML/NeuralPDE.jl/issues/406#issuecomment-964474831). This appear to occur when there is a rapidly decreasing function (like exp()) in the integral.

I don't know if we need some regularization to get most infinite interval integrals to work, or if we simply need to build the representation based on the transformed integral in Quadrature.jl. But I think this is a bug. 

Any thoughts on how to proceed?

**Comment by ChrisRackauckas** [comment]
More transformations 😅

**Comment by killah-t-cell** [comment]
@ChrisRackauckas how do you mean? 😄

**Comment by ChrisRackauckas** [comment]
We can symbolically transform the equations by doing a substitution `t -> f(tau)` where f is a chosen transformation to a finite domain, similar to the quadrature.

**Comment by killah-t-cell** [comment]
Gotcha. Started work on this here https://github.com/SciML/NeuralPDE.jl/pull/436. Still figuring out what the symbolic transformation should look like, but making progress.

**Comment by ChrisRackauckas** [comment]
You should just be able to take the equation and do `substitute(t => f(tau))`

**Comment by killah-t-cell** [comment]
But you’d still need to define the transformation f, no? f would transform the variables according to this  https://github.com/stevengj/cubature#infinite-intervals. Or am I misunderstanding?

**Comment by ChrisRackauckas** [comment]
Yes exactly.

**Comment by killah-t-cell** [comment]
Hm yeah that makes sense. The inner representation looks like this in this case. It looks better, but it failed at `solve` with a `τ is not defined`. Does this look right to you?

```julia
(Expr[:((cord, var"##θ#257", phi, derivative, integral, u, p)->begin
          begin
              (var"##θ#2571",) = (var"##θ#257"[1:353],)
              (phi1,) = (phi[1],)
              let (t, x, v) = (fill((/)(τ, (+)(1, (*)(-1, (^)(τ, 2)))), size(cord[[1], :])), cord[[1], :], cord[[2], :])
                  begin
                      cord1 = vcat(t, x, v)
                  end
                  integral(u, cord1, phi, [1], RuntimeGeneratedFunctions.RuntimeGeneratedFunction{(:cord, Symbol("##θ#257"), :phi, :derivative, :integral, :u, :p), NeuralPDE.var"#_RGF_ModTag", NeuralPDE.var"#_RGF_ModTag",
...
```

**Comment by ChrisRackauckas** [comment]
The transform is now automated.
